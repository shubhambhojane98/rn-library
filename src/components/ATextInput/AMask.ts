export const CreditCard = [
    /\d/,
    /\d/,
    /\d/,
    /\d/,
    ' ',
    [/\d/],
    [/\d/],
    [/\d/],
    [/\d/],
    ' ',
    [/\d/],
    [/\d/],
    [/\d/],
    [/\d/],
    ' ',
    /\d/,
    /\d/,
    /\d/,
    /\d/,
  ];
  
  export const PhoneNo = [
    '(',
    /\d/,
    /\d/,
    ')',
    ' ',
    /\d/,
    /\d/,
    /\d/,
    /\d/,
    /\d/,
    '-',
    /\d/,
    /\d/,
    /\d/,
    /\d/,
  ];
  
  export const ZipCodeMask = [/\d/, /\d/, /\d/, /\d/, /\d/, '-', /\d/, /\d/, /\d/];
  
  export const DateMask = [/\d/, /\d/, ' / ', /\d/, /\d/, ' / ',  /\d/, /\d/, /\d/, /\d/];
  