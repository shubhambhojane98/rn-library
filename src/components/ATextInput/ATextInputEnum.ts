export enum TextAlignVertical {
    Center = 'center',
    Top = 'top',
    Auto = 'auto',
    Bottom = 'bottom',
  }
  
  export enum AutoCapitalize {
    None = 'none',
    Sentences = 'sentences',
    Words = 'words',
    Characters = 'characters',
  }
  