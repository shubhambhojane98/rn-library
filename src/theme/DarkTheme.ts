import type {Theme} from '../utils/types';

let DarkTheme: Theme;

DarkTheme = {
  dark: true,
  colors: {
    primary: '#BB86FC',
    primaryVariant: '#3700B3',
    secondary: '#03DAC6',
    secondaryVariant: '#018786',
    background: '#262833',
    error: '#CF6679',
    textColor: '#FFF',
    shadowColor: '#D7D7D7',
    black: '#000',
    white: '#FFF',
    lightgrey: '#D7D7D7',
  },
};

export default DarkTheme;
