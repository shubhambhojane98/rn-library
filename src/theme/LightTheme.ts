import type {Theme} from '../utils/types';

let LightTheme: Theme;

LightTheme = {
  dark: false,
  colors: {
    primary: '#6200EE',
    primaryVariant: '#3700B3',
    secondary: '#03DAC6',
    secondaryVariant: '#018786',
    background: '#FFF',
    error: '#B00020',
    textColor: '#000',
    shadowColor: '#222',
    black: '#000',
    white: '#FFF',
    lightgrey: '#D7D7D7',
    backgroundLayout: 'rgba(0, 0, 0, 0.8)',
    transparent: 'transparent',
    linkColor: '#0000EE',
    greyOpac50: 'rgba(53, 53, 53, 0.5)',
    greyOpac10: 'rgba(53, 53, 53, 0.1)',
  },
};

export default LightTheme;
