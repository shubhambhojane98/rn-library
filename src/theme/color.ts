import {Appearance} from 'react-native';

interface colors {
  yellow: string;
  black: string;
  red: string;
  violet: string;
  taupe: string;
  beige: string;
  grey: string;
  lightgrey: string;
  blue: string;
  darkGrey: string;
  white: string;
  transportation: string;
  logistic: string;
  environment: string;
  sports: string;
  healthcare: string;
  background: string;
  textColor: string;
  placeholderTextColor: string;
  switchBackgroundColor: string;
  paginationDotColor: string;
  paginationDotBgColor: string;
  blackOpac80: string;
  blackOpac60: string;
  blackOpac40: string;
  blackOpac20: string;
  greyOpac80: string;
  greyOpac60: string;
  greyOpac50: string;
  greyOpac40: string;
  greyOpac20: string;
  greyOpac10: string;
  switchActiveBgColor: string;
  switchInActiveBgColor: string;
  switchKnobColor: string;
}

const primaryColors = {
  yellow: '#FAD000',
  black: '#231F20',
  red: '#A81E23',
  violet: '#533467',
  taupe: '#695958',
  beige: '#C3B59F',
  grey: '#353535',
  white: '#fff',
  lightgrey: '#D7D7D7',
  blue: '#007BC1',
  darkGrey: 'rgba(0, 0, 0, 0.2)',
};

const opacColors = {
  blackOpac80: 'rgba(0, 0, 0, 0.8)',
  blackOpac60: 'rgba(0, 0, 0, 0.6)',
  blackOpac40: 'rgba(0, 0, 0, 0.4)',
  blackOpac20: 'rgba(0, 0, 0, 0.2)',
  greyOpac80: 'rgba(53, 53, 53, 0.8)',
  greyOpac60: 'rgba(53, 53, 53, 0.6)',
  greyOpac50: 'rgba(53, 53, 53, 0.5)',
  greyOpac40: 'rgba(53, 53, 53, 0.4)',
  greyOpac20: 'rgba(53, 53, 53, 0.2)',
  greyOpac10: 'rgba(53, 53, 53, 0.1)',
};

const categoryColors = {
  transportation: '#007BC1',
  logistic: '#EF4065',
  environment: '#009A46',
  sports: '#F26524',
  healthcare: '#00ABBE',
};

let themeColors: colors;

const colorScheme = Appearance.getColorScheme();
if (colorScheme === 'dark') {
  // Use dark color scheme
  themeColors = {
    ...primaryColors,
    ...categoryColors,
    ...opacColors,
    background: primaryColors.black,
    textColor: primaryColors.white,
    placeholderTextColor: '#5D5D5D',
    switchBackgroundColor: primaryColors.white,
    paginationDotColor: primaryColors.white,
    paginationDotBgColor: primaryColors.darkGrey,
    switchActiveBgColor: primaryColors.yellow,
    switchInActiveBgColor: primaryColors.white,
    switchKnobColor: primaryColors.black,
  };
} else {
  // Use light color scheme
  themeColors = {
    ...primaryColors,
    ...categoryColors,
    ...opacColors,
    background: primaryColors.yellow,
    textColor: primaryColors.black,
    placeholderTextColor: '#5D5D5D',
    switchBackgroundColor: primaryColors.lightgrey,
    paginationDotColor: primaryColors.white,
    paginationDotBgColor: primaryColors.darkGrey,
    switchActiveBgColor: primaryColors.yellow,
    switchInActiveBgColor: primaryColors.white,
    switchKnobColor: primaryColors.black,
  };
}

export default themeColors;
