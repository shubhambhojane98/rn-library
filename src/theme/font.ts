import { Platform, PlatformOSType } from 'react-native';
import { Fonts } from '../utils/types';


export default function configureFonts(config?: {
    [platform in PlatformOSType | 'default']?: Fonts;
  }): Fonts {
    const fonts = Platform.select({ ...config }) as Fonts;
    return fonts;
  }