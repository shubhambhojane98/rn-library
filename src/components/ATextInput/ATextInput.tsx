import React, {FC, useState} from 'react';
import {
  View,
  TextInput,
  StyleSheet,
  KeyboardTypeOptions,
  ReturnKeyTypeOptions,
  I18nManager,
} from 'react-native';
import {moderateScale} from 'react-native-size-matters';
import IconSVG from '../../assets/svgs';
import {Color} from '../../theme';
import {TextAlignVertical, AutoCapitalize} from './ATextInputEnum';
import ATypography from '../ATypography/ATypography';
import MaskInput from 'react-native-mask-input';
import {TypographyVariant} from '../ATypography/ATypographyEnum';

interface Props {
  placeholder?: string;
  placeholderTextColor?: string;
  marginTop?: number;
  marginBottom?: number;
  marginLeft?: number;
  marginRight?: number;
  underlineColorAndroid?: string;
  borderColor?: string;
  value?: string | any | undefined;
  autoFocus?: boolean;
  keyboardType?: KeyboardTypeOptions;
  textContentType?: string | any;
  onChangeText?: (Name: any, masked?: any, unmasked?: any) => void;
  autoCorrect?: boolean;
  returnKeyType?: ReturnKeyTypeOptions | undefined;
  textAlignVertical?: TextAlignVertical;
  onSubmitEditing?: (event: any) => void;
  error?: boolean;
  errorText?: string;
  multiline?: boolean;
  noExteriorView?: boolean;
  secureTextEntry?: boolean;
  autoCapitalize?: AutoCapitalize;
  maxLength?: number | undefined;
  onEndEditing?: any;
  selectionColor?: any;
  isPassword?: any;
  width?: any;
  disable?: boolean;
  iconHeight?: number;
  iconWidth?: number;
  editable?: boolean;
  rightIcon?: any;
  leftIcon?: any;
  isTopText?: boolean;
  showObfuscatedValue?: boolean;
  placeholderFillCharacter?: string;
  obfuscationCharacter?: string;
  mask?: any;
  inputRef?: any;
}

const defaultSize = 20;
const defaultMargin = 5;

const ATextInput: FC<Props> = ({
  placeholder,
  placeholderTextColor,
  marginTop = defaultMargin,
  marginBottom = defaultMargin,
  marginLeft = defaultMargin,
  marginRight = defaultMargin,
  borderColor,
  value,
  underlineColorAndroid,
  autoFocus,
  keyboardType,
  textContentType,
  onChangeText,
  autoCorrect,
  returnKeyType,
  iconHeight = defaultSize,
  iconWidth = defaultSize,
  textAlignVertical,
  editable,
  onSubmitEditing,
  multiline,
  errorText,
  autoCapitalize,
  maxLength,
  isPassword,
  onEndEditing,
  selectionColor,
  width,
  rightIcon,
  leftIcon,
  disable,
  showObfuscatedValue,
  obfuscationCharacter,
  placeholderFillCharacter,
  mask,
  inputRef,
}) => {
  const [visible, setVisible] = useState(true);
  const [isFocused, setIsFocused] = useState(false);
  const [secureTextEntry, setSecureTextEntry] = useState(false);

  const onFocus = () => {
    setIsFocused(true);
  };

  const onBlur = () => {
    setIsFocused(false);
  };
  const handlePassword = () => {
    setVisible(!visible);
    setSecureTextEntry(!secureTextEntry);
  };

  return (
    <View>
      <View>
        {disable && (
          <View
            style={{
              ...styles.disableTextInputContainer,
              width: width,
              marginTop: moderateScale(marginTop),
              marginBottom: moderateScale(marginBottom),
              marginLeft: moderateScale(marginLeft),
              marginRight: moderateScale(marginRight),
            }}>
            <TextInput
              style={styles.disableTextinput}
              placeholder={placeholder}
              editable={false}
              placeholderTextColor={
                placeholderTextColor || Color.placeholderTextColor
              }
            />
            {rightIcon ? (
              <View style={styles.icon}>
                <IconSVG name={rightIcon} />
              </View>
            ) : null}
          </View>
        )}

        {!mask && !disable && (
          <View
            style={[
              styles.textinputContainer,
              isFocused && {...styles.shadow, borderColor: Color.black},
              !isFocused && {
                borderColor: errorText ? Color.red : borderColor,
              },
              {
                marginTop: moderateScale(marginTop),
                marginBottom: moderateScale(marginBottom),
                marginLeft: moderateScale(marginLeft),
                marginRight: moderateScale(marginRight),
                width: width,
              },
            ]}>
            {!isPassword && leftIcon && (
              <View style={styles.icon}>
                <IconSVG
                  height={moderateScale(iconHeight)}
                  width={moderateScale(iconWidth)}
                  name={leftIcon}
                />
              </View>
            )}
            <TextInput
              style={styles.textinput}
              placeholder={placeholder}
              onChangeText={onChangeText}
              editable={editable}
              ref={inputRef}
              autoCorrect={autoCorrect}
              placeholderTextColor={
                placeholderTextColor || Color.placeholderTextColor
              }
              value={value}
              autoFocus={autoFocus}
              keyboardType={keyboardType}
              textContentType={textContentType}
              underlineColorAndroid={underlineColorAndroid}
              textAlignVertical={textAlignVertical}
              returnKeyType={returnKeyType}
              secureTextEntry={secureTextEntry}
              autoCapitalize={autoCapitalize}
              multiline={multiline}
              onSubmitEditing={onSubmitEditing}
              onEndEditing={onEndEditing}
              onFocus={onFocus}
              onBlur={onBlur}
              maxLength={maxLength}
              selectionColor={selectionColor}
            />
            {isPassword && (
              <View style={styles.icon}>
                <IconSVG
                  height={iconHeight}
                  onPress={handlePassword}
                  name={visible ? 'showpassword' : 'hidepassword'}
                />
              </View>
            )}
            {!isPassword && rightIcon && (
              <View style={styles.icon}>
                <IconSVG
                  height={moderateScale(iconHeight)}
                  width={moderateScale(iconWidth)}
                  name={rightIcon}
                />
              </View>
            )}
          </View>
        )}
        {mask && (
          <View
            style={[
              styles.textinputContainer,
              isFocused && {...styles.shadow, borderColor: Color.black},
              !isFocused && {
                borderColor: errorText ? Color.red : borderColor,
              },
              {
                marginTop: moderateScale(marginTop),
                marginBottom: moderateScale(marginBottom),
                marginLeft: moderateScale(marginLeft),
                marginRight: moderateScale(marginRight),
                width: width,
              },
            ]}>
            <MaskInput
              placeholder={placeholder}
              style={styles.textinput}
              value={value}
              ref={inputRef}
              showObfuscatedValue={showObfuscatedValue}
              obfuscationCharacter={obfuscationCharacter}
              placeholderFillCharacter={placeholderFillCharacter}
              onChangeText={onChangeText}
              onSubmitEditing={onSubmitEditing}
              onEndEditing={onEndEditing}
              onFocus={onFocus}
              onBlur={onBlur}
              keyboardType={keyboardType}
              autoCapitalize={autoCapitalize}
              textAlignVertical={textAlignVertical}
              mask={mask}
            />
          </View>
        )}
      </View>
      <View
        style={{
          marginLeft: moderateScale(marginLeft),
          marginRight: moderateScale(marginRight),
        }}>
        {errorText && !disable && !isFocused ? (
          <ATypography
            children={errorText}
            variant={TypographyVariant.PRIMARY}
            color={Color.red}
            fontSize={14}
          />
        ) : null}
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  textinputContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    borderRadius: 2,
    borderWidth: 1,
    backgroundColor: Color.white,
  },
  icon: {
    paddingRight: moderateScale(15, 0.2),
    paddingLeft: moderateScale(5, 0.2),
  },
  textinput: {
    flex: 1,
    height: moderateScale(48, 0.2),
    textAlign: I18nManager.isRTL ? 'right' : 'left',
    paddingLeft: moderateScale(15, 0.2),
    color: Color.black,
  },
  shadow: {
    shadowColor: Color.yellow,
    shadowOffset: {
      width: 0,
      height: 0,
    },
    shadowOpacity: 0.2,
    shadowRadius: 3,
    elevation: 15,
  },
  disableTextInputContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    backgroundColor: Color.lightgrey,
    borderRadius: 2,
  },
  disableTextinput: {
    flex: 1,
    height: moderateScale(48, 0.2),
    paddingLeft: moderateScale(15, 0.2),
    textAlign: I18nManager.isRTL ? 'right' : 'left',
  },
});

export default ATextInput;
