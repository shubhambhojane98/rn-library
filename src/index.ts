import {defaultScale} from './utils/Common';
import {consoleLog, consoleWarn, consoleInfo} from './utils/Log';
import {CheckboxStatus} from './components/ACheckbox/ACheckboxEnum';
import {Color} from './theme';
export {default as AButton} from './components/AButton/AButton';
export {default as ATypography} from './components/ATypography/ATypography';
export {default as AAvatar} from './components/AAvatar/AAvatar';
export {default as ABadge} from './components/ABadge/ABadge';
export {default as AImage} from './components/AImage/AImage';
export {default as ARadioButton} from './components/ARadioButton/ARadioButton';
export {default as ARadioButtonGroup} from './components/ARadioButtonGroup/ARadioButtonGroup';
export {default as ATextInput} from './components/ATextInput/ATextInput';
export {default as ASwitch} from './components/ASwitch/ASwitch';
export {default as AVirtualizedList} from './components/AVirtualizedList/AVirtualizedList';
export {default as AModal} from './components/AModal/AModal';
export {default as ASwitchSelector} from './components/ASwitchSelector/ASwitchSelector';
export {default as AStarRating} from './components/AStarRating/AStarRating';
export {default as ACheckbox} from './components/ACheckbox/ACheckbox';
export {default as AToast} from './components/AToast/AToast';
export {default as ACard} from './components/ACard/ACard';
export {consoleLog, consoleWarn, consoleInfo};
export {Color};
export {CheckboxStatus};
