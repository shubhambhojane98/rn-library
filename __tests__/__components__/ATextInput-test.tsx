import 'react-native';
import React, {useState} from 'react';
import renderer from 'react-test-renderer';
import ATextInput from '../../src/components/ATextInput/ATextInput';
import {Color} from '../../src/theme';
import {CreditCard} from '../../src/components/ATextInput/AMask';

it('renders correctly default prop', () => {
  const tree = renderer
    .create(<ATextInput placeholder="placeholder" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with default & margingTop prop', () => {
  const tree = renderer
    .create(
      <ATextInput placeholder="placeholder" marginTop={10} disable={true} />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with default & margingTop prop', () => {
  const tree = renderer
    .create(
      <ATextInput
        errorText={'Error text'}
        placeholder="Placeholder"
        marginTop={10}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with isPassword & secureTextEntry prop', () => {
  const tree = renderer
    .create(
      <ATextInput
        placeholder="Placeholder"
        isPassword={true}
        secureTextEntry={true}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with rightIcon & iconHeight & iconWidth & borderColor & keyboardType prop', () => {
  const tree = renderer
    .create(
      <ATextInput
        placeholder="Placeholder"
        keyboardType="name-phone-pad"
        rightIcon={'gifticon'}
        iconHeight={35}
        iconWidth={35}
        borderColor={Color.blue}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with mask & value  prop', () => {
  const tree = renderer
    .create(
      <ATextInput mask={CreditCard} value={jest.fn} onChangeText={jest.fn} />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with mask & value & errorText  prop', () => {
  const tree = renderer
    .create(
      <ATextInput
        mask={CreditCard}
        errorText={'Error Text'}
        value={jest.fn}
        onChangeText={jest.fn}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
