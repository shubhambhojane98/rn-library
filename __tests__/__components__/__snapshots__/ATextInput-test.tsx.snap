// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly default prop 1`] = `
<View>
  <View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRadius": 2,
            "borderWidth": 1,
            "flexDirection": "row",
          },
          false,
          Object {
            "borderColor": undefined,
          },
          Object {
            "marginBottom": 7.857142857142857,
            "marginLeft": 7.857142857142857,
            "marginRight": 7.857142857142857,
            "marginTop": 7.857142857142857,
            "width": undefined,
          },
        ]
      }
    >
      <TextInput
        onBlur={[Function]}
        onFocus={[Function]}
        placeholder="placeholder"
        placeholderTextColor="#5D5D5D"
        secureTextEntry={false}
        style={
          Object {
            "color": "#231F20",
            "flex": 1,
            "height": 58.971428571428575,
            "paddingLeft": 18.428571428571427,
            "textAlign": "left",
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "marginLeft": 7.857142857142857,
        "marginRight": 7.857142857142857,
      }
    }
  />
</View>
`;

exports[`renders correctly with default & margingTop prop 1`] = `
<View>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#D7D7D7",
          "borderRadius": 2,
          "flexDirection": "row",
          "marginBottom": 7.857142857142857,
          "marginLeft": 7.857142857142857,
          "marginRight": 7.857142857142857,
          "marginTop": 15.714285714285714,
          "width": undefined,
        }
      }
    >
      <TextInput
        editable={false}
        placeholder="placeholder"
        placeholderTextColor="#5D5D5D"
        style={
          Object {
            "flex": 1,
            "height": 58.971428571428575,
            "paddingLeft": 18.428571428571427,
            "textAlign": "left",
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "marginLeft": 7.857142857142857,
        "marginRight": 7.857142857142857,
      }
    }
  />
</View>
`;

exports[`renders correctly with default & margingTop prop 2`] = `
<View>
  <View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRadius": 2,
            "borderWidth": 1,
            "flexDirection": "row",
          },
          false,
          Object {
            "borderColor": "#A81E23",
          },
          Object {
            "marginBottom": 7.857142857142857,
            "marginLeft": 7.857142857142857,
            "marginRight": 7.857142857142857,
            "marginTop": 15.714285714285714,
            "width": undefined,
          },
        ]
      }
    >
      <TextInput
        onBlur={[Function]}
        onFocus={[Function]}
        placeholder="Placeholder"
        placeholderTextColor="#5D5D5D"
        secureTextEntry={false}
        style={
          Object {
            "color": "#231F20",
            "flex": 1,
            "height": 58.971428571428575,
            "paddingLeft": 18.428571428571427,
            "textAlign": "left",
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "marginLeft": 7.857142857142857,
        "marginRight": 7.857142857142857,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Mulish-Regular",
            "fontSize": 17.2,
            "fontWeight": "normal",
          },
          Object {
            "color": "#A81E23",
          },
          Object {
            "textAlign": "left",
          },
          undefined,
        ]
      }
    >
      Error text
    </Text>
  </View>
</View>
`;

exports[`renders correctly with isPassword & secureTextEntry prop 1`] = `
<View>
  <View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRadius": 2,
            "borderWidth": 1,
            "flexDirection": "row",
          },
          false,
          Object {
            "borderColor": undefined,
          },
          Object {
            "marginBottom": 7.857142857142857,
            "marginLeft": 7.857142857142857,
            "marginRight": 7.857142857142857,
            "marginTop": 7.857142857142857,
            "width": undefined,
          },
        ]
      }
    >
      <TextInput
        onBlur={[Function]}
        onFocus={[Function]}
        placeholder="Placeholder"
        placeholderTextColor="#5D5D5D"
        secureTextEntry={false}
        style={
          Object {
            "color": "#231F20",
            "flex": 1,
            "height": 58.971428571428575,
            "paddingLeft": 18.428571428571427,
            "textAlign": "left",
          }
        }
      />
      <View
        style={
          Object {
            "paddingLeft": 6.142857142857142,
            "paddingRight": 18.428571428571427,
          }
        }
      >
        <SvgMock
          height={20}
          name="showpassword"
          onPress={[Function]}
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginLeft": 7.857142857142857,
        "marginRight": 7.857142857142857,
      }
    }
  />
</View>
`;

exports[`renders correctly with mask & value  prop 1`] = `
<View>
  <View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRadius": 2,
            "borderWidth": 1,
            "flexDirection": "row",
          },
          false,
          Object {
            "borderColor": undefined,
          },
          Object {
            "marginBottom": 7.857142857142857,
            "marginLeft": 7.857142857142857,
            "marginRight": 7.857142857142857,
            "marginTop": 7.857142857142857,
            "width": undefined,
          },
        ]
      }
    >
      <TextInput
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        style={
          Object {
            "color": "#231F20",
            "flex": 1,
            "height": 58.971428571428575,
            "paddingLeft": 18.428571428571427,
            "textAlign": "left",
          }
        }
        value=""
      />
    </View>
  </View>
  <View
    style={
      Object {
        "marginLeft": 7.857142857142857,
        "marginRight": 7.857142857142857,
      }
    }
  />
</View>
`;

exports[`renders correctly with mask & value & errorText  prop 1`] = `
<View>
  <View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRadius": 2,
            "borderWidth": 1,
            "flexDirection": "row",
          },
          false,
          Object {
            "borderColor": "#A81E23",
          },
          Object {
            "marginBottom": 7.857142857142857,
            "marginLeft": 7.857142857142857,
            "marginRight": 7.857142857142857,
            "marginTop": 7.857142857142857,
            "width": undefined,
          },
        ]
      }
    >
      <TextInput
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        style={
          Object {
            "color": "#231F20",
            "flex": 1,
            "height": 58.971428571428575,
            "paddingLeft": 18.428571428571427,
            "textAlign": "left",
          }
        }
        value=""
      />
    </View>
  </View>
  <View
    style={
      Object {
        "marginLeft": 7.857142857142857,
        "marginRight": 7.857142857142857,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Mulish-Regular",
            "fontSize": 17.2,
            "fontWeight": "normal",
          },
          Object {
            "color": "#A81E23",
          },
          Object {
            "textAlign": "left",
          },
          undefined,
        ]
      }
    >
      Error Text
    </Text>
  </View>
</View>
`;

exports[`renders correctly with rightIcon & iconHeight & iconWidth & borderColor & keyboardType prop 1`] = `
<View>
  <View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderRadius": 2,
            "borderWidth": 1,
            "flexDirection": "row",
          },
          false,
          Object {
            "borderColor": "#007BC1",
          },
          Object {
            "marginBottom": 7.857142857142857,
            "marginLeft": 7.857142857142857,
            "marginRight": 7.857142857142857,
            "marginTop": 7.857142857142857,
            "width": undefined,
          },
        ]
      }
    >
      <TextInput
        keyboardType="name-phone-pad"
        onBlur={[Function]}
        onFocus={[Function]}
        placeholder="Placeholder"
        placeholderTextColor="#5D5D5D"
        secureTextEntry={false}
        style={
          Object {
            "color": "#231F20",
            "flex": 1,
            "height": 58.971428571428575,
            "paddingLeft": 18.428571428571427,
            "textAlign": "left",
          }
        }
      />
      <View
        style={
          Object {
            "paddingLeft": 6.142857142857142,
            "paddingRight": 18.428571428571427,
          }
        }
      >
        <SvgMock
          height={55}
          name="gifticon"
          width={55}
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginLeft": 7.857142857142857,
        "marginRight": 7.857142857142857,
      }
    }
  />
</View>
`;
